### PR TITLE
records: add logic to migrate drafts

### PR DESCRIPTION
**Pull request by ppanero**
- closes https://github.com/zenodo/zenodo-rdm/issues/241

### CODE REVIEW

File: invenio_rdm_migrator/streams/records/table_generators/records.py [hunk 1]
```diff
@@ -102,102 +54,90 @@ def __init__(self, parent_cache, communities_cache):
...
-            # parent in cache - update version
-            cached_parent = self.parent_cache[parent["json"]["id"]]
-            # check if current record is a new version of the cached one
-            if cached_parent["version"]["latest_index"] < rec["index"]:
-                cached_parent["version"] = dict(
-                    latest_index=rec["index"], latest_id=rec["id"]
-                )
         # record files
-        record_files = data["record_files"]
+        record_files = record["record_files"]
```
Comment (zzacharo) [inline invenio_rdm_migrator/streams/records/table_generators/records.py:140]: if I understand correctly, that is wrong as record is set to `transformed_entry["record"]` which doesn't have a `record_files` property...

File: invenio_rdm_migrator/streams/cache.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record streams cache."""
+
+
+class ParentCache:
```
Comment (ppanero) [inline invenio_rdm_migrator/streams/cache.py:11]: we had some inconsistencies and I wanted to enforce the cache to be in a usable state so I put checks on `add`

**Comment by ppanero** [comment]
As discussed in Discord Im merging this and will perform changes from review/comments in a subsequent PR.

**Review by zzacharo — COMMENTED** [review]
minor comments

**Pull request merged**